frontend/src/pages/user/admin/allCoupons/[adminCouponID].tsx

```diff
@@ -0,0 +1,196 @@
+import { useState } from 'react';
+import { useParams } from 'react-router-dom';
+import { Row, Col } from 'react-bootstrap';
+import DatePicker from 'react-datepicker';
+import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
+import { faPen, faTrash } from '@fortawesome/free-solid-svg-icons';
+import 'react-datepicker/dist/react-datepicker.css';
+
+import TButton from '@components/TButton';
+import InfoItem from '@components/InfoItem';
+import CouponItem from '@components/CouponItem';
+import couponData from '@pages/coupon/couponData.json';
+
+interface Props {
+  id: number | null;
+  name: string;
+  policy: string;
+  date: string;
+  introduction: string;
+  tags: { name: string }[];
+}
+
+const EachAdminCoupon = () => {
+  const params = useParams();
+  const id = params.coupon_id;
+
+  const data: Props = {
+    id: null,
+    name: '',
+    policy: '',
+    date: '',
+    introduction: '',
+    tags: [],
+  };
+
+  const foundCoupon = couponData.find((coupon) => coupon.id.toString() === id);
+
+  if (foundCoupon) {
+    Object.assign(data, foundCoupon);
+  }
+
+  const tagStyle = {
+    borderRadius: '30px',
+    background: ' var(--button_light)',
+    padding: '1% 1% 1% 3%',
+    color: 'white',
+    margin: '5px 0 5px 5px',
+    width: '100%',
+  };
+
+  const [tag, setTag] = useState('');
+  const [tagContainer, setTagContainer] = useState<string[]>(data.tags.map((tag) => tag.name));
+  const [modification, setModification] = useState<boolean[]>(Array(data.tags.length).fill(false));
+  const [name, setName] = useState<string>(data.name);
+  const [policy, setPolicy] = useState<string>(data.policy);
+  const [introduction, setIntroduction] = useState<string>(data.introduction);
+  const [date, setDate] = useState<string>(data.date);
+
+  const addNewTag = (event: React.KeyboardEvent<HTMLInputElement>) => {
+    if (event.keyCode === 229) return;
```

Comment:
What does this keycode do? If not possible to not use magic numbers please leave a comment

Reply:
It's to deal with undefined behalvire. [this address it](https://github.com/facebook/react/issues/14512).

Reply:
You can link this issue in the comment